front/components/assistant/details/tabs/AgentInfoTab/AssistantToolsSection.tsx

```diff
@@ -158,10 +168,70 @@ export function AssistantToolsSection({
   );
 }
 
+/**
+ * Hook to fetch available toolsets for an agent configuration.
+ * Only fetches data if the agent has a toolsets action configured.
+ */
+function useAvailableToolsets({
```

Comment:
I decided to put the hook inline because it won't be used elsewhere and I'm guessing it might be burned with skills -> this will increase the chances of it being cleaned up 🙏🏻